request.js

```diff
@@ -74,14 +74,14 @@
             request.onload = function(){
                 if((request.statusText === 'OK' && request.status === 200) || typeof request.statusText === 'undefined'){
                     methods.success.apply(methods, utils.parse(request));
-                    methods.always.apply();    
+                    methods.always.apply();
                 } else {
                     methods.error.apply(methods, utils.parse(request));
                     methods.always.apply();
                 }
             };
-            request.onerror = function(){
-                methods.error.apply(methods, utils.parse(request));
+            request.onerror = function(e){
+                methods.error.apply(methods, e);
```

Comment:
I understand, but that wasn't my point. I meant that if you send `request` as first parameter of the `.apply`, we can access the object on the callback.

1- Sending `callbacks`:
```javascript
.get(url).error(function(){
    console.log(this); // {success:..., error:..., always:...}
}
```

2- Sending `request`:
```javascript
.get(url).error(function(){
    console.log(this); // {status:....,statusText:....,response:....}
}
```

Just an idea.